os/board/rtl8730e/src/rtl8730e_st7701.c

```diff
@@ -0,0 +1,208 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/lcd/st7701.h>
+#include <tinyara/mipidsi/mipi_dsi.h>
+#include "objects.h"
+#include "PinNames.h"
+#include "gpio_api.h"
+
+static void rtl8730_st7701_lcd_init(void);
+static void rtl8730_st7701_gpio_reset(void);
+static void rtl8730_st7701_cache_invalidate(u32 *buffer, int size);
+static void rtl8730_st7701_lcd_reload(void);
+static void rtl8730_st7701_lcd_enable(void);
+static void rtl8730_st7701_lcd_layer_enable(int layer, bool enable);
+static void rtl8730_st7701_lcd_put_area(u8 *lcd_img_buffer, u32 x1, u32 y1, u32 x2, u32 y2);
+static void rtl8730_st7701_enable_lcdc(void);
+FAR void mipidsi_mode_switch(bool do_enable);
+FAR struct mipi_dsi_host *mipi_dsi_host_initialize(struct lcd_data *config);
+FAR struct mipi_dsi_device *mipi_dsi_device_register(FAR struct mipi_dsi_host *host, FAR const char *name, int channel);
+FAR struct lcd_dev_s *st7701_lcdinitialize(FAR struct mipi_dsi_device *dsi, struct st7701_config_s *config);
+
+struct st7701_config_s g_rtl8730_st7701_config_s = {
+	.init = rtl8730_st7701_lcd_init,
+	.reset = rtl8730_st7701_gpio_reset,
+	.cache_invalidate = rtl8730_st7701_cache_invalidate,
+	.lcd_reload = rtl8730_st7701_lcd_reload,
+	.lcd_enable = rtl8730_st7701_lcd_enable,
+	.lcd_layer_enable = rtl8730_st7701_lcd_layer_enable,
+	.lcd_put_area = rtl8730_st7701_lcd_put_area,
+};
+
+LCDC_TypeDef *pLCDC = LCDC;
+LCDC_InitTypeDef lcdc_init_struct;
+
+u8 *lcd_img_buffer1 = (u8 *)(DDR_BASE + LCDC_IMG_BUF_OFFSET2 + LCDC_IMG_BUF_SIZE);
+
+static void LcdcInitValues(struct lcd_data config)
+{
+	LCDC_StructInit(&lcdc_init_struct);
+	lcdc_init_struct.LCDC_ImageWidth = config.XPixels;
+	lcdc_init_struct.LCDC_ImageHeight = config.YPixels;
+	for (u8 idx = 0; idx < LCDC_LAYER_MAX_NUM; idx++) {
+		lcdc_init_struct.layerx[idx].LCDC_LayerImgFormat = LCDC_LAYER_IMG_FORMAT_RGB565;
+		lcdc_init_struct.layerx[idx].LCDC_LayerImgBaseAddr = (u32) lcd_img_buffer1;
+		lcdc_init_struct.layerx[idx].LCDC_LayerHorizontalStart = 1;
+		lcdc_init_struct.layerx[idx].LCDC_LayerHorizontalStop = config.XPixels;
+		lcdc_init_struct.layerx[idx].LCDC_LayerVerticalStart = 1;
+		lcdc_init_struct.layerx[idx].LCDC_LayerVerticalStop = config.YPixels;
+		lcdc_init_struct.layerx[idx].LCDC_LayerColorKeyingEn = ENABLE;
+	}
+}
+
+static u32 LCDCRgbHvIsr(void *Data)
```

Comment:
This irq handler is not registered in anywhere. 
Shall we register this in rtl8730_st7701_lcd_init?

Reply:
yes, this irq was registered in original file, but I didnt' see any usecase for this. The LCD is working fine without this irq.
Instead there is another irq handler in amebasmart_mipi.c file which seems useful .
But since you mentioned it, I am rechecking this irq handler from documentation again.